platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/BasicSoftwareVersion.java

```diff
@@ -108,29 +108,9 @@ public int compareTo(final SoftwareVersion that) {
         }
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public boolean equals(final Object o) {
-        if (this == o) {
-            return true;
-        }
-
-        if (!(o instanceof BasicSoftwareVersion that)) {
-            return false;
-        }
-
-        return softwareVersion == that.softwareVersion;
```

Comment:
Would it be better to implement it but throw an exception if you want to force us to not use it anywhere?

Reply:
`BasicSoftwareVersion` was created with the idea that some applications may choose to use it directly as opposed to implementing their own software version. Even though this is used for testing, it's not considered test code, and so adding logic designed for tests seems like the wrong thing to do.